allensdk/brain_observatory/behavior/trials_processing.py

```diff
@@ -215,37 +215,121 @@ def get_trial_reward_time(rebased_reward_times, start_time, stop_time):
-    assert not (go==True and catch==True), "both `go` and `catch` cannot be True, they are mutually exclusive categories"
-    assert not (go==True and auto_rewarded==True), "both `go` and `auto_rewarded` cannot be True, they are mutually exclusive categories"
+
+def get_trial_timing(
+        event_dict: dict, stimulus_presentations_df: pd.DataFrame,
+        licks: List[float], go: bool, catch: bool, auto_rewarded: bool,
+        hit: bool, false_alarm: bool, aborted: bool):
+    """
+    Extract a dictionary of trial timing data.
+    See trial_data_from_log for a description of the trial types.
+
+    Parameters
+    ==========
+    event_dict: dict
+        Dictionary of trial events in the well-known `pkl` file
+    stimulus_presentations_df: pd.DataFrame
+        pandas dataframe of stimulus presentations, from the
+        `get_stimulus_presentations` response for the
+        BehaviorOphysSession.api.
+    licks: List[float]
+        list of lick timestamps, from the `get_licks` response for
+        the BehaviorOphysSession.api.
+    go: bool
+        True if "go" trial, False otherwise. Mutually exclusive with
+        `catch`.
+    catch: bool
+        True if "catch" trial, False otherwise. Mutually exclusive
+        with `go.`
+    auto_rewarded: bool
+        True if "auto_rewarded" trial, False otherwise.
+    hit: bool
+        True if "hit" trial, False otherwise
+    false_alarm: bool
+        True if "false_alarm" trial, False otherwise
+    aborted: bool
+        True if "aborted" trial, False otherwise
+
+    Returns
+    =======
+    dict
+        start_time: float
+            The time the trial started (in seconds elapsed from
+            recording start)
+        stop_time: float
+            The time the trial ended (in seconds elapsed from
+            recording start)
+        trial_length: float
+            Duration of the trial in seconds
+        response_time: float
+            The response time, for non-aborted trials. This is equal
+            to the first lick in the trial. For aborted trials or trials
+            without licks, `response_time` is NaN.
+        change_frame: int
+            The frame number that the stimulus changed
+        change_time: float
+            The time in seconds that the stimulus changed
+        response_latency: float or None
+            The time in seconds between the stimulus change and the
+            animal's lick response, if the trial is a "go", "catch", or
+            "auto_rewarded" type. If the animal did not respond,
+            return `float("inf")`. In all other cases, return None.
+
+    Notes
+    =====
+    The following parameters are mutually exclusive (exactly one can
+    be true):
+        hit, miss, false_alarm, aborted, auto_rewarded
+    """
+    assert not (aborted and (hit or false_alarm or auto_rewarded)), (
```

Comment:
Updated this for flake8, the diff is weird but the assertions weren't changed (except the first one is new, was added).